### PR TITLE
Link comments with pins

### PR DESCRIPTION
**Pull request by noahmarxx**
The only real issues left here are that the POST request for /pins/id/comments creates two comments in the comments collection, and the GET route for /pins/id/comments should return a JSON array off comment objects, while right now it just displays the objectIDs of each comment.

### CODE REVIEW

File: backend/backend.js [hunk 1]
```diff
@@ -45,6 +46,12 @@ app.get("/pins", async (req, res) => {
   }
 });
 
+app.get("/pins/:id", async (req, res) => {
```
Comment (davidayu) [inline backend/backend.js:49]: We made drastic changes to the get function. Can you pull from main and refactor the code to match it?

Comment (davidayu) [reply backend/backend.js:49]: Nevermind.

File: backend/models/pin-services.js [hunk 2]
```diff
@@ -84,6 +85,43 @@ async function downvotePin(id) {
...
+    pin.comments.pull(commentID);
+    const savedPin = await pin.save();
+    return savedPin;
+  } catch (error) {
+    console.log(error);
+    return undefined;
+  }
+}
+
+async function getPinComments(id) {
```
Comment (davidayu) [inline backend/models/pin-services.js:114]: Can you change this to return a list of json objects rather than a list of commentIds?

Comment (ksittner) [reply backend/models/pin-services.js:114]: You might be able to just change line 117 to be
pin = await pinModel.findById(id).populate('comments');

I recommend using MongoDB's population functionality: [](https://mongoosejs.com/docs/populate.html)

**Pull request closed without merging**